src/Futhark/CodeGen/Backends/GenericC.hs

```diff
@@ -644,7 +644,9 @@ setMem dest src space = do
                                                $string:src_s) != 0) {
                        return 1;
                      }|]
-    else stm [C.cstm|$exp:dest = $exp:src;|]
+    else case space of
+           ScalarSpace{} -> stm [C.cstm|$exp:dest[0] = $exp:src[0];|]
```

Comment:
I don't think this is right.  You need a loop to copy every element of the array.  We probably don't yet generate code that needs that, though.

Reply:
What if it's a loop with multiple dimensions, should I have a nested loop?

Reply:
The number of elements in the array, which is given by the dimensions of the ScalarSpace.